typings/built-in-component-factories.d.ts

```diff
@@ -0,0 +1,126 @@
+import {
+  WithOptionalTheme,
+  ThemedStyledFunction,
+} from './styled-components'
+
+type HTMLStyledComponentFactory<Theme, Props, Element> = ThemedStyledFunction<
+  Props,
+  Theme,
+  WithOptionalTheme<Props, Theme>
+>;
+
+export interface HTMLComponentFactory<Theme, Props extends { theme?: Theme; }> {
```

Comment:
I don't see how is the type used anywhere in the tests and in the main typings? the file has not been imported anywhere.